allauth/socialaccount/providers/notion/tests.py

```diff
@@ -0,0 +1,120 @@
+import base64
+import hashlib
+import requests
+from urllib.parse import parse_qs, urlparse
+
+from django.urls import reverse
+from django.utils.http import urlencode
+
+from allauth.socialaccount import app_settings
+from allauth.socialaccount.tests import OAuth2TestsMixin
+from allauth.tests import MockedResponse, TestCase, mocked_response
+
+from .provider import NotionProvider
+
+
+class NotionTests(OAuth2TestsMixin, TestCase):
+    provider_id = NotionProvider.id
+
+    def get_mocked_response(self):
+        return MockedResponse(
+            200,
+            """
+            {
+                "workspace_id": "workspace-abc",
+                "owner": {
+                    "user": {
+                        "id": "test123",
+                        "name": "John Doe",
+                        "avatar_url": "",
+                        "person": {"email": "john@example.com"},
+                    }
+                },
+            }
+            """,
+        )  # noqa
+
+    def get_login_response_json(self, with_refresh_token=False):
+        """
+        Docs here:
+        https://developers.notion.com/docs/authorization#step-4-notion-responds-with-an-access_token-and-additional-information
+        """
+        return """
+        {
+            "access_token": "test123",
+            "bot_id": "bot-abc",
+            "duplicated_template_id": "template-abc",
+            "owner": {
+            "workspace_id": "workspace-abc",
+                "user": {
+                    "id": "test123",
+                    "name": "John Doe",
+                    "avatar_url": "",
+                    "person": {
+                        "email": "john@example.com"
+                    }
+                }
+            },
+            "workspace_icon": "https://example.com/icon.png",
+            "workspace_id": "workspace-abc",
+            "workspace_name": "My Workspace"
+        }
+        """
+
+    def login(self, resp_mock=None, process="login", with_refresh_token=True):
+        resp = self.client.post(
+            reverse(self.provider.id + "_login")
+            + "?"
+            + urlencode(dict(process=process))
+        )
+
+        p = urlparse(resp["location"])
+        q = parse_qs(p.query)
+
+        # Always false.
+        pkce_enabled = False
```

Comment:
**Question**: There are a few `if pkce_enabled` down below, but here it is hardcoded to false.  Is that intentional?

Reply:
Notion doesn't support PKCE. Looked around when I first implemented this, but did not find a decent way to properly disabled its test in the old codebase, so brought over the `login` test and manually set `pkce_enabled` to `False` with minor mods to get it working.

I see that the `OAuthTestsMixin.login` has been modified in the last few weeks, and just tried 

```python
class NotionTests(OAuth2TestsMixin, TestCase):
    provider_id = NotionProvider.id
    pkce_enabled_default = False
```

with no luck either due to `OAuth2TestMixin.test_provider_has_pkce_params` which overwrites `self.provider.pkce_enabled_default`. Unsure if this is a bug or intended. 

Whatever the case, just committed a new change that cleans up the function a bit more and removes `pkce_enabled` entirely, while also adding `pkce_enabled_default = False` as a class variable. If you think there are any better way to go about this do let me know!